boards/posix/native_posix/hw_models_top.c

```diff
@@ -125,6 +126,10 @@ void hwm_find_next_timer(void)
 void hwm_main_loop(void)
 {
 	while (1) {
+#ifdef CONFIG_HAS_SDL
+		hwm_sdl_poll_events();
```

Comment:
Not goody. (if you place code here you have no control about how often it will be called)
If you want your code to be a HW model, and to be called periodically (say every 30 simulated ms), add a timer similar to `hw_timer_timer`:
* Add a new value in next_timer_index 
* Add your a pointer to your timer in Timer_list, and a call to you function in hwm_main_loop().
* Be sure to update your timer (`+=30e3`) into a future time value each time your function is called.